Source/WebCore/html/SearchInputType.cpp

```diff
@@ -199,12 +202,12 @@ float SearchInputType::decorationWidth() const
 
 void SearchInputType::setValue(const String& sanitizedValue, bool valueChanged, TextFieldEventBehavior eventBehavior, TextControlSetValueSelection selection)
 {
-    bool emptinessChanged = valueChanged && sanitizedValue.isEmpty() != element()->value().isEmpty();
+    bool emptinessChanged = valueChanged && sanitizedValue.isEmpty() != protectedElement()->value().isEmpty();
 
     BaseTextInputType::setValue(sanitizedValue, valueChanged, eventBehavior, selection);
 
-    if (m_cancelButton && emptinessChanged)
-        m_cancelButton->invalidateStyleInternal();
+    if (RefPtr cancelButton = m_cancelButton; cancelButton && emptinessChanged)
```

Comment:
Let’s do a separate early return for !emptinessChanged here before we RefPtr, since Speedometer shows this code can be hot, and it’s a free optimization.